### DISCUSSION TITLE
umask being ignored

### DISCUSSION THREAD

**Discussion by rstanonik** [versity/versitygw#877] [Q&A]
We're building versitygw on ubuntu22
```
go version
go version go1.23.2 linux/amd64

./versitygw --version
Version  : v1.0.7
Build    : fed72e9
BuildTime: 2024-10-02_04:15:00PM
```

We'd like newly created bucket directories to be 775; ie, group writable. 
We "umask 002" before starting versitygw, but it seems to ignore the umask and directories are 755.
```
umask 002
./versitygw --access XXX --secret XXX posix /mnt/versity

```

Then from another machine
`aws --profile versitygw --endpoint-url http://XXX.ucsd.edu:7070 s3 mb s3://first`

The resulting directory is 755
```
ls -ld /mnt/versitygw/first
drwxr-xr-x 2 root root 4096 Oct  7 17:03 /mnt/versitygw/first
```

I've searched documentation, discussions, and google generally for keywords versitygw and umask, but haven't found any advice.

Any suggestions?
Thanks,
Ron

**Comment by benmcclelland** [comment]
We can open a feature request for this. The 755 is hard coded in there for now, but we can add an option to set this:
https://github.com/versity/versitygw/blob/d2b0d24520fd899525d1f3ff0831d5099262248f/backend/posix/posix.go#L274-L277

**Comment by benmcclelland** [comment]
opened #878

**Comment by rstanonik** [comment]
Thanks!

**Comment by rstanonik** [comment]
Can the file default perms also be configurable; eg, 664?
Thanks again,
Ron

**Reply by benmcclelland** [reply]
I'll look into the options here. I think the default permissions are usually set by directory umask, but since we are using O_TMPFILE and linking into the namespace for some filesystems, we might not be getting the directory umask behavior in those cases.
